samples/net/latmon/src/main.c

```diff
@@ -0,0 +1,718 @@
+	}
+
+	ret = gpio_pin_configure_dt(&pulse, GPIO_OUTPUT_HIGH);
+	if (ret < 0) {
+		LOG_ERR("failed configuring pulse pin");
+		return ret;
+	}
+
+	ret = gpio_pin_configure_dt(&ack, GPIO_INPUT);
+	if (ret < 0) {
+		LOG_ERR("failed configuring ack pin");
+		return ret;
+	}
+
+	ret = gpio_pin_interrupt_configure_dt(&ack, GPIO_INT_EDGE_RISING);
+	if (ret < 0) {
+		LOG_ERR("failed configuring ack pin interrupt");
+		return ret;
+	}
+
+	gpio_init_callback(&gpio_cb, gpio_ack_handler, BIT(ack.pin));
+
+	ret = gpio_add_callback_dt(&ack, &gpio_cb);
+	if (ret < 0) {
+		LOG_ERR("failed adding ack pin callback");
+		return ret;
+	}
+
+	if (gpio_is_ready_dt(&led_run))
+		 gpio_pin_configure_dt(&led_run, GPIO_OUTPUT_INACTIVE);
+
+	if (gpio_is_ready_dt(&led_wait))
+		 gpio_pin_configure_dt(&led_wait, GPIO_OUTPUT_INACTIVE);
+
+	if (gpio_is_ready_dt(&led_dhcp))
+		 gpio_pin_configure_dt(&led_dhcp, GPIO_OUTPUT_INACTIVE);
+
+	return ret;
+}
+
+static ssize_t write_latmus_socket(const void *buf, size_t count)
+{
+	ssize_t total_written = 0;
+	ssize_t bytes_written;
+
+	while (count > 0) {
+		const char *p = (const char *)buf + total_written;
+		bytes_written = send(latmus_socket ,p ,count, 0);
+		if (bytes_written < 0) {
+			if (errno == EINTR)
+				continue;
+			return -1;
+		}
+		if (!bytes_written)
+			break;
+
+		total_written += bytes_written;
+		count -= bytes_written;
+	}
+
+    return total_written;
+}
+
+static void reset_sampling_counters(void)
+{
+	current_samples = 0;
+	min_lat = UINT_MAX;
+	overruns = 0;
+	sum_lat = 0;
+	max_lat = 0;
+}
+
+static int send_sample_data(void)
+{
+	struct latmon_net_data ndata = { };
+
+	ndata.sum_lat_hi = htonl(sum_lat >> 32);
+	ndata.sum_lat_lo = htonl(sum_lat & 0xffffffff);
+	ndata.min_lat = htonl(min_lat);
+	ndata.max_lat = htonl(max_lat);
+	ndata.overruns = htonl(overruns);
+	ndata.samples = htonl(current_samples);
+
+	reset_sampling_counters();
+
+	return (write_latmus_socket(&ndata, sizeof(ndata)) <= 0 ? -1 : 0);
+}
+
+static int send_trailing_data(void)
+{
+	int count = latmus_cells * sizeof(histogram[0]);
+	ssize_t ret = 0;
+	int cell = 0;
+
+	if (current_samples && send_sample_data() < 0)
+		return -1;
+
+	if (send_sample_data() < 0)
+		return -1;
+
+	for (cell = 0; cell < latmus_cells; cell++)
+		histogram[cell] = htonl(histogram[cell]);
+
+	ret = write_latmus_socket(histogram, count);
+	if (ret < 0) {
+		LOG_INF("failed tx histogram (ret=%d, errno %d)", ret, errno);
+		memset(histogram, 0, sizeof(histogram));
+		return -1;
+	}
+
+	return 0;
+}
+
+static void reset_ack_event(void)
+{
+	k_sem_reset(&ack_event);
+	atomic_clear(&ack_wait);
+}
+
+static int wait_for_ack(bool *warm)
+{
+	static uint32_t cnt = 0;
+
+	if (k_sem_take(&ack_event, K_SECONDS(1)) == 0) {
+		if (*warm)
+			return 0;
+
+		if (++cnt < max_samples_per_bulk)
+			return -1;
+
+		*warm = true;
+		cnt = 0;
+
+		return 0;
+	}
+
+	LOG_ERR("timeout waiting for ack ");
+
+	if (!atomic_clear(&ack_wait))
+		LOG_ERR("unexpected value clearing ack_wait ");
```

Comment:
Always put curly braces around `if` blocks.

```suggestion
	if (!atomic_clear(&ack_wait)) {
		LOG_ERR("unexpected value clearing ack_wait ");
	}
```

Reply:
ok